### ISSUE TITLE
Anyway to check if an item with specific index match the query?

### ISSUE THREAD

**Issue by Losses** [blevesearch/bleve#1429]
I'm facing such a context:
* Have some predefined queries;
* Have some newly indexed items.

What I want to do is: 
If these newly added item could be matched by the predefined queries.

I've searched a lot from the document but could not figure out an efficient way to do that, need some help...

**Comment by mschoch** [comment]
So, I don't have any really good answer, but I'll share some ideas.  Basically, bleve is focused on building a index (and spending resources to do it), with the goal that over time we benefit from this by being able to answer arbitrary queries.  The use-case you've described here is sort of inverted, you have the queries over a long time, and want to know if arbitrary new documents would match the query.

But, there are techniques to try and do what you want.  One example I know of is the Couchbase integration of bleve indexes with their query language.  In some cases, they have some expression, and they need to feed in one-row-at-a-time from their query execution output, and see if that row would satisfy the expression.  The technique they use is to build an in-memory only index that only ever contains one document.  Then they can query against that in-memory index, and do a fast drop of the index data, and start over with the next document.  You could try something similar, when the newly added items come in, toss them into a separate in-memory only index as well, then run the queries you want against it, and then drop it when you're done querying.

Here are some links to the Couchbase code I mentioned, I'm not going to suggest it's obvious, but I'm happy to try and explain what they're doing if you have questions trying to replicate this approach.

1.  They aren't using Bleve's built-in in-memory indexes, instead they create a traditional bleve index, using the Moss key-value store, and using an in-memory only mode of it.  They ran some benchmarks and concluded this was faster.
2.  Another benefit of using Moss, was that it offered a fast way to drop all the existing data (not something offered by the built-in in-memory index)

Here is the code where they build the in-memory index:

https://github.com/couchbase/n1fty/blob/bc5cf396b1a623dd182da4d3320bfe1d300f438f/verify/verify_result.go#L141-L182

And here they add a document to the index, run the query, then reset the contents of the index: 

https://github.com/couchbase/n1fty/blob/bc5cf396b1a623dd182da4d3320bfe1d300f438f/verify/verify_result.go#L245-L270

So, it's not exactly simple, but it is possible to do something similar.  If you interested in trying this approach and have more questions, please post them here.

**Comment by Losses** [comment]
Thank you for the answer, I tried another approach days ago and want to discuss it with you, I don't know if I'm misusing something or finish the task in a stupid way.

What I did is while indexing the data, add a new `Field` calls `BatchTaskTImeStamp`, data indexed at the same batch share a common time stamp, then search matches with `Facet`, which limit the search result only with the time stamp I just provided.

Is this a reasonable way to finish the task?

**Comment by mschoch** [comment]
That approach will work as well.  I think I assumed you had already tried that and it wasn't performing well enough.

**Comment by Losses** [comment]
Yep... but my dataset is not very big so the performance is... acceptable?
